packages/picasso.js/src/core/chart-components/labels/strategies/rows.js

```diff
@@ -125,12 +125,11 @@ export function rows({ settings, chart, nodes, renderer, style }, placer = place
       continue;
     }
 
-    let totalHeight = 0;
+    let measuredHeight = 0;
     let measurements = [];
     let texts = [];
 
-    let maxHeight = type === 'circle' ? 2 * bounds.r * CIRCLE_FACTOR : bounds.height;
-    totalHeight += rowSettings.padding;
+    let boundHeight = type === 'circle' ? 2 * bounds.r * CIRCLE_FACTOR : bounds.height;
     let j;
     for (j = 0; j < labelSettings.length; j++) {
       let lblStngs = labelSettings[j];
```

Comment:
[L114](https://github.com/qlik-oss/picasso.js/blob/12ac80df3a7f1c0851daa3b30c39101301169ba1/packages/picasso.js/src/core/chart-components/labels/strategies/rows.js#L114) refers to `labelSettings` which is as same as `rowSettings`. The code needs to be refactored, otherwise I think row padding should be considered once per node.